yt_dlp/extractor/youtube/_base.py

```diff
@@ -595,6 +601,31 @@ def _get_sid_authorization_header(self, origin='https://www.youtube.com', user_s
 
         return ' '.join(authorizations)
 
+    @property
+    def is_authenticated(self):
+        return self._has_auth_cookies
+
+    @property
+    def _has_auth_cookies(self):
+        yt_sapisid, yt_1psapisid, yt_3psapisid = self._get_sid_cookies()
+        # YouTube doesn't appear to clear 3PSAPISID when rotating cookies (as of 2025-04-26)
+        # But LOGIN_INFO is cleared and should exist if logged in
+        has_login_info = 'LOGIN_INFO' in self._youtube_cookies
+        return bool(has_login_info and (yt_sapisid or yt_1psapisid or yt_3psapisid))
+
+    def _request_webpage(self, *args, **kwargs):
+        response = super()._request_webpage(*args, **kwargs)
+
+        # Check that we are still logged-in and cookies have not rotated after every request
+        if self._had_cookie_auth and not self._has_auth_cookies:
+            self.report_warning(
```

Comment:
Just thinking we could keep `is_authenticated` as a `cached_property` if we do this:
```suggestion
        if self._had_cookie_auth and not self._has_auth_cookies:
            if self.is_authenticated:
                del self.is_authenticated
            self.report_warning(
```

Reply:
kinda hacky though and bound to the implementation of it

Reply:
True. And if the user passed cookies, we'll be invoking `_has_auth_cookies` after every request anyways even if we do cache it, so it's not like caching is that useful